### PR TITLE
Upgrade to use version 3.X of the AWS provider

### PR DESCRIPTION
**Pull request by jsf9k**
## 🗣 Description

This pull request updates this repository to use version 3.x of the AWS Terraform provider.

See also cisagov/freeipa-server-tf-module#36.

## 💭 Motivation and Context

Version 3.X of the AWS provider has been out for a while now and has stabilized.  We should upgrade before we get left behind.

## 🧪 Testing

I successfully ran a `terraform apply` for this repository in our staging COOL environment using these changes and those in cisagov/freeipa-server-tf-module#36.  No resources needed to be changed.

## 🚥 Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (causes existing functionality to change)
- [x] Update Terraform provider

## ✅ Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by jsf9k** [comment]
Note that the check failures should be resolved once cisagov/freeipa-server-tf-module#36 is merged.

File: freeipa.tf [hunk 1]
```diff
@@ -31,7 +31,7 @@ module "security_groups" {
 
 # Create the IPA servers
 module "ipa0" {
-  source = "github.com/cisagov/freeipa-server-tf-module"
+  source = "github.com/cisagov/freeipa-server-tf-module?ref=improvement%2Fupdate-to-aws-3.0"
```
Comment (dav3r) [inline freeipa.tf]: Standard reminder to switch back to the main branch later. 👍

**Review by dav3r — APPROVED** [review]
Looks great to me! 🎸

**Review by mcdonnnj — APPROVED** [review]
Nice work! :shipit: 🚀

**Pull request merged**